src/crimson/net/ProtocolV1.h

```diff
@@ -26,11 +26,11 @@ class ProtocolV1 final : public Protocol {
 
   void trigger_close() override;
 
-  seastar::future<> write_message(MessageRef msg) override;
-
-  seastar::future<> do_keepalive() override;
-
-  seastar::future<> do_keepalive_ack() override;
+  ceph::bufferlist do_sweep_messages(
+      const std::deque<MessageRef>& msgs,
+      size_t num_msgs,
+      bool require_keepalive,
+      bool require_keepalive_ack) override;
```

Comment:
might want to take this opportunity to use `final`.

Reply:
I thought the class `ProtocolV1` was already marked `final`.
Please see discussion https://github.com/ceph/ceph/pull/26710#discussion_r270933077